mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -433,18 +433,23 @@ struct ChainedTensorCast : public OpRewritePattern<CastOp> {
 
     // We can remove the intermediate cast if joining all three produces the
     // same result as just joining the source and result shapes.
-    auto firstJoin =
-        joinShapes(joinShapes(sourceType, intermediateType), resultType);
+    auto firstJoin = joinShapes(sourceType, intermediateType);
 
     // The join might not exist if the cast sequence would fail at runtime.
     if (!firstJoin)
       return failure();
 
+    auto secondJoin = joinShapes(firstJoin, resultType);
+
+    // The join might not exist if the cast sequence would fail at runtime.
+    if (!secondJoin)
```

Comment:
```suggestion
    if (!(firstJoin || secondJoin))
     return failure()
```

Reply:
The `!firstJoin` condition is already checked in the code above, and we can't merge both the conditions since the secondJoin will happen only if the firstJoin is a non-Null value.

Reply:
Makes sense. Thanks.